crates/avro/src/encode.rs

```diff
@@ -397,6 +402,7 @@ mod test {
             "m2_no_addl": {"d": true},
             "m2_disallowed_field": {"not valid": true},
             "zz - ex": ["extra", 3],
+            "int_with_zero_frac": serde_json::Value::Number(serde_json::Number::from_f64(1.0).unwrap()),
```

Comment:
nit: i'm pretty sure this can just be 1.0

Reply:
.. Huh, you're right. I could have sworn I tried that in the Dekaf tests and it didn't trigger the encoding error 🤔 
Anyway, changed!